src/main/java/org/cyclonedx/CycloneDxSchema.java

```diff
@@ -271,6 +271,8 @@ private Schema getXmlSchema16() throws SAXException {
 
   public Schema getXmlSchema(InputStream... inputStreams) throws SAXException {
     final SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+    schemaFactory.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");
```

Comment:
does this actually prevent XEE? does not look like this, looks like exfiltration is disabled, but XEE are still evaluated, or are they?

what about `XMLConstants.FEATURE_SECURE_PROCESSING` ? 
what about `schemaFactory.setExpandEntityReferences(false);`? 
or similar?

Reply:
`schemaFactory.setExpandEntityReferences` doesn't exist.

The `ACCESS_EXTERNAL_*` values are a list of protocols which are allowed, setting it to empty makes it disallow all:

* https://docs.oracle.com/javase/8/docs/api/javax/xml/XMLConstants.html#ACCESS_EXTERNAL_SCHEMA
* https://docs.oracle.com/javase/8/docs/api/javax/xml/XMLConstants.html#ACCESS_EXTERNAL_DTD

I verified that this also prevents `file://` access.

I will add `FEATURE_SECURE_PROCESSING` though, thanks for the tip.